kernel-rpm-script

```diff
@@ -0,0 +1,302 @@
+#!/bin/sh
+
+op=$(echo "$0" | sed -e "s/.*-//")
+
+while true ; do
+    case $1 in
+	--version)
+	    version="$2"
+	    shift 2
+	    ;;
+	--release)
+	    release="$2"
+	    shift 2
+	    ;;
+	--name)
+	    name="$2"
+	    shift 2
+	    ;;
+
+	--kernelrelease)
+	    kernelrelease="$2"
+	    shift 2
+	    ;;
+	--flavor)
+	    flavor="$2"
+	    shift 2
+	    ;;
+	--variant)
+	    flavor="$2"
+	    shift 2
+	    ;;
+
+	--usrmerged)
+	    usrmerged="$2"
+	    shift 2
+	    ;;
+	--image)
+	    image="$2"
+	    shift 2
+	    ;;
+	--systemmap)
+	    systemmap="$2"
+	    shift 2
+	    ;;
+
+	--modulesdir)
+	    modulesdir="$2"
+	    shift 2
+	    ;;
+	--basepackage)
```

Comment:
This is tested in almost every step below. I think we should provide separate scripts for basepackages and other packages.

Reply:
Large part of the script does not depend on basepackage, though. That would get duplicated. Also this is what is in the kernel now, splitting it at this time is needlessly hairy.

Reply:
but if you really want separate basepackage script is should be done now so that the API defines separate scripts

Reply:
Double-checking the part that doesn't depend on `basepackage` ...

1.  the aarch64/ PAGE_SIZE issue in `%pre`. I'm quite positive that this check should really depend on `basepackage`. `-extra` depends on the base pkg anyway, so if the latter can't be installed, the former won't be installable either.
 2.  `trigger_purge_kernels` in `%post` - unsure why that would be run by a non-base package. Same for the remval of `/boot/do_purge_kernels` in `%postun`.
 3. the `$usrmerged` stuff in `%post` - definitely only necessary for the base package.
 4. bootloader installation in `%post` - again, I see no point running that for a non-base package. Even worse for `%postun` - why remove a bootloader entry for `kernel-default` just because `kernel-default-extra` is being removed?
 5. `run_wm2 --remove-kernel` in `%postun` - I'm quite surprised to see this being called for non-basepackages. This is ancient code from 2011 that has never been changed in `postun.sh` - weird. Looks like a bug to me.

Reply:
Maybe we should use kernel-inkmp-script instead?

Reply:
> Maybe we should use kernel-inkmp-script instead?

I'm not sure I understand this remark correctly. We have 3 types of packages currently, kernel basepackages, kernel addon packages (like `-extra`), and KMPs. Each type gets different treatment by the scripts. I think we should leave it at that.

Reply:
And the difference between kernel-default-extra and kernel-default-dlm-kmp is what exactly?

Reply:
Excellent question. None, I would say. If I say "KMP", I mean real 3rd-party KMPs that install into `updates/` (or `extra/`), and for which we create `weak-updates` symlinks.

Reply:
The point is the wm2 invocation:

 * type 1 (basepackage): `--add-kernel`
 * type 2 (kernel extra): `--add-kernel-modules`
 * type 3 (KMP): `--add-kmp`

and `--remove-...`, respectively.

Reply:
the difference is INITRD_IN_POSTTRANS=1 for kernel-default-dlm-kmp but not kernel-default-extra

Reply:
... which will be subject to change.

Reply:
That difference between `-extra` and the "kmp" packages doesn't make any sense to me.

Reply:
Neither to me so I switched -extra to the inkmp script. I suppose that the difference is mostly due to historical reasons that make the to scripts completely separate in kernel-source.

Reply:
To be on the safe side, I think we should take some of the issues we encountered to the kernel ML. I'll prepare a post.